### DISCUSSION TITLE
Mercure Support

### DISCUSSION THREAD

**Discussion by sirinath** [activej/activej#176] [Ideas]
Can you add [mercure](https://github.com/dunglas/mercure) support. 

This is used by [hotwire](https://hotwire.dev/)  turbo.

**Comment by eduard-vasinskyi** [comment]
Hello, @sirinath
Mercure seems to be an application-specific protocol. Its implementation is out of scope of ActiveJ framework.

However, I believe a Mercure protocol can be implemented using ActiveJ HTTP [server](https://github.com/activej/activej/blob/master/core-http/src/main/java/io/activej/http/AsyncHttpServer.java) and [client](https://github.com/activej/activej/blob/master/core-http/src/main/java/io/activej/http/AsyncHttpClient.java) as a basis. Our HTTP module is flexible enough for this task as it gives the user full control of HTTP headers/cookies, etc. This allows for the effortless implementation of custom HTTP protocols on top of the [ActiveJ HTTP module](https://github.com/activej/activej/tree/master/core-http)
